lib/src/source/git.dart

```diff
@@ -40,6 +41,49 @@ class GitSource extends CachedSource {
     });
   }
 
+  /// Gets the list of all versions from git tags.
+  /// Anything that Version.parse understands is considered a version,
+  /// it will also attempt to strip off a preceding 'v' e.g. v1.0.0
+  Future<List<Pubspec>> getVersions(String name, description) async {
+    if (!_useVersionTags(description)) {
+      // No need to get versions if the useVersionTags option is false
+      return super.getVersions(name, description);
+    }
+
+    PackageId id =
+        new PackageId(name, this.name, null, _getDescription(description));
```

Comment:
Passing `null` for the version here is a smell. It looks like `_repoCachePath` doesn't actually require a version, so it should be modified to take a `PackageRef` instead of a `PackageId`. You can update other call sites by calling `PackageId.toRef()` to get the ref.